module-services/service-desktop/ServiceDesktop.cpp

```diff
@@ -52,44 +31,90 @@ ServiceDesktop::~ServiceDesktop()
+    usbSecurityModel = std::make_unique<sdesktop::USBSecurityModel>(this, settings.get());
+
+    connectHandler<db::NotificationMessage>();
+    connectHandler<locks::UnlockedPhone>();
+    connectHandler<locks::LockedPhone>();
+    connectHandler<locks::NextPhoneUnlockAttemptLockTime>();
+    connectHandler<message::bluetooth::ResponseStatus>();
+    connectHandler<message::bluetooth::ResponseBondedDevices>();
+    connectHandler<message::bluetooth::ResponseVisibleDevices>();
+    connectHandler<sdesktop::developerMode::DeveloperModeRequest>();
+    connectHandler<sdesktop::BackupMessage>();
+    connectHandler<sdesktop::RestoreMessage>();
+    connectHandler<sdesktop::FactoryMessage>();
+    connectHandler<sdesktop::usb::USBConfigured>();
+    connectHandler<sdesktop::usb::USBDisconnected>();
+    connectHandler<sevm::BatteryStatusChangeMessage>();
+
+    checkChargingCondition();
+
+    return sys::ReturnCodes::Success;
 }
 
-auto ServiceDesktop::generateDeviceUniqueId() -> void
+sys::ReturnCodes ServiceDesktop::DeinitHandler()
 {
-    const auto deviceUniqueId = utils::generateRandomId(sdesktop::DeviceUniqueIdLength);
-    LOG_SENSITIVE(LOGINFO, "Device unique id: %s", deviceUniqueId.c_str());
-    setDeviceUniqueId(deviceUniqueId);
+    return usbWorkerDeinit();
 }
 
-auto ServiceDesktop::setDeviceUniqueId(const std::string &token) -> void
+sys::ReturnCodes ServiceDesktop::SwitchPowerModeHandler(const sys::ServicePowerMode /*mode*/)
 {
-    return settings->setValue(sdesktop::DeviceUniqueIdName, token);
+    return sys::ReturnCodes::Success;
 }
 
-auto ServiceDesktop::getDeviceToken() -> std::string
+sys::MessagePointer ServiceDesktop::DataReceivedHandler(sys::DataMessage * /*msg*/, sys::ResponseMessage *resp)
 {
-    std::string tokenSeed = getDeviceUniqueId();
-
-    if (tokenSeed.empty()) {
-        LOG_DEBUG("Device unique id is empty, generating one...");
-        generateDeviceUniqueId();
-        tokenSeed = getDeviceUniqueId();
+    auto response = std::make_shared<sys::ResponseMessage>();
+    if (resp == nullptr) {
+        return response;
+    }
+    if (resp->responseTo != MessageType::DBQuery) {
+        return response;
+    }
+    if (auto queryResponse = dynamic_cast<db::QueryResponse *>(resp)) {
+        auto result = queryResponse->getResult();
+        if (result == nullptr) {
+            LOG_ERROR("Wrong result - nullptr!");
+            return response;
+        }
+        if (result->hasListener()) {
+            LOG_DEBUG("Handling result...");
+            result->handle();
+        }
     }
 
-    return tokenSeed;
+    return response;
 }
 
-auto ServiceDesktop::getDeviceUniqueId() const -> std::string
+std::string ServiceDesktop::prepareBackupFilename()
 {
-    return settings->getValue(sdesktop::DeviceUniqueIdName);
+    std::array<char, 64> backupFileName{};
+    std::time_t now;
+    std::time(&now);
+    std::strftime(backupFileName.data(), backupFileName.size(), "%FT%OH%OM%OSZ", std::localtime(&now));
```

Comment:
maybe it's worth moving the format text to constexpr and write unit test for it?

Reply:
For now, I'm living this as is, but I'll consider changing this in the next iteration of service-desktop refactor.